src/MoBi.Engine/Sbml/UnitDefinitionImporter.cs

```diff
@@ -103,30 +120,29 @@ public IDimension ConvertUnit(UnitDefinition unitDefinition)
             return dimension;
          }
 
+         var combinedUnit = new CombinedUnit();
          for (long i = 0; i < unitDefinition.getNumUnits(); i++)
          {
             var sbmlUnitDefinition = unitDefinition.getUnit(i);
-            var mobiUnit = SBMLBaseUnitToMoBi(sbmlUnitDefinition);
-            if (mobiUnit == null) continue;
-            var unitDimension = _moBiDimensionFactory.DimensionForUnit(mobiUnit.Name);
-            if (unitDimension != Constants.Dimension.NO_DIMENSION)
-            {
-               _sbmlInformation.MobiDimension[sbmlUnit] = unitDimension;
-               _unitConvertionDictionary.Add(sbmlUnit, new UnitConvertionInfo() { Dimension = unitDimension, Unit = mobiUnit, Rate = Math.Pow(sbmlUnitDefinition.getMultiplier(), sbmlUnitDefinition.getExponent()) });
-               return unitDimension;
-            }
+            combinedUnit.AddUnit(
+               sbmlUnitDefinition.getKind(), 
+               sbmlUnitDefinition.getExponent(), 
+               sbmlUnitDefinition.getMultiplier(),
+               _baseUnitsDictionary);
          }
+         var unitName = combinedUnit.Name;
+         var unitDimension = _moBiDimensionFactory.DimensionForUnit(unitName);
```

Comment:
` public (IDimension dimension, Unit unit) FindUnit(string unitName)` returns the dimension and unit... as an FYI

Reply:
Good to know.